lib/cog/messages.ex

```diff
@@ -84,6 +86,18 @@ defmodule Cog.Messages.Command do
 
   # Information about the Cog user invoking the command
   field :user, :map, required: true
+
+  @doc "Creates a new Command from an invocation, options, and args"
+  defmacro create(invocation, options, args) do
+    quote bind_quoted: [invocation: invocation, options: options, args: args] do
+      %Cog.Messages.Command{command: invocation.meta.full_command_name,
+                            options: options,
+                            args: args,
+                            invocation_id: invocation.id,
+                            services_root: ServiceEndpoint.url()}
```

Comment:
It was to clean up the code a bit without obscuring intent. It was originally a function but due to how `Conduit` does code generation it had to be a macro.